include/rnp/rnp2.h

```diff
@@ -0,0 +1,283 @@
+* @param msg the message to sign
+* @param msg_len the length of msg in bytes
+* @param sig on success, the output signature buffer
+* @param sig_len on success, the length of sig in bytes
+*/
+rnp_result_t rnp_sign(rnp_keyring_t keyring,
+                      const char *  ident,
+                      const char *  hash_fn,
+                      bool          clearsign,
+                      bool          armor,
+                      const uint8_t msg[],
+                      size_t        msg_len,
+                      uint8_t **    sig,
+                      size_t *      sig_len);
+
+/**
+* Generate an embedded signature (the output will include the
+* message contents)
+* @param keyring the keyring
+* @param ident the key to sign with
+* @param hash_fn the hash function to use
+* @param msg the message to sign
+* @param msg_len the length of msg in bytes
+* @param sig on success, the output signature buffer
+* @param sig_len on success, the length of sig in bytes
+*/
+rnp_result_t rnp_sign_detached(rnp_keyring_t keyring,
+                               const char *  ident,
+                               const char *  hash_fn,
+                               bool          armor,
+                               const uint8_t msg[],
+                               size_t        msg_len,
+                               uint8_t **    sig,
+                               size_t *      sig_len);
+
+/**
+* Verify a signature with embedded message
+* @param key the keyring
+* @param sig the signature
+* @param sig_len length of signature in bytes
+* @param msg on succes, output buffer to the message
+* @param msg_len on success, length of msg in bytes
+*/
+
+rnp_result_t rnp_verify(
+  rnp_keyring_t keyring, const uint8_t sig[], size_t sig_len, uint8_t **msg, size_t *msg_len);
+
+/**
+* Verify a detached signature
+* @param key the key to verify with
+* @param msg the message bits being verified
+* @param msg_len length of msg in bytes
+* @parma sig the signature
+* @param sig_len length of signature in bytes
+* @return RNP_SUCCESS if valid otherwise error
+*/
+rnp_result_t rnp_verify_detached(rnp_keyring_t keyring,
+                                 const uint8_t msg[],
+                                 size_t        msg_len,
+                                 const uint8_t sig[],
+                                 size_t        sig_len);
+
+/**
+* Verify a detached signature
+* @param key the key to verify with
+* @param file_path path to the file to be verified
+* @parma sig the signature
+* @param sig_len length of signature in bytes
+* @return RNP_SUCCESS if valid otherwise error
+*/
+rnp_result_t rnp_verify_detached_file(rnp_keyring_t keyring,
+                                      const char *  file_path,
+                                      const uint8_t sig[],
+                                      size_t        sig_len);
+
+/* Encryption/decryption operations */
+
+rnp_result_t rnp_encrypt(rnp_keyring_t keyring,
```

Comment:
Same as with sign() call - what about multiple recipients?

Reply:
Same deal, we don't really support it internally yet, so until then...

Reply:
Actually, now I started moving pk-encryption/decryption into the stream architecture, and plan to add multiple-recipient encryption during that. So, soon we'll support it :-)

Reply:
@ronaldtse Is this something Ribose will need in the initial FFI/ruby-rnp? (If not, then we can leave this for later)

Reply:
Well it is easy to support at the API level for now, pass `const char* receipients[], size_t recipients_len` and for now just error if more than 1 is specified.